examples/crust_node.rs

```diff
@@ -0,0 +1,214 @@
+                       .filter(|node| node.endpoint == new_node.endpoint)
+                       .count() == 0 &&
+       self.our_eps.iter()
+                   .filter(|& our_ep| our_ep == &new_node.endpoint)
+                   .count() == 0 {
+      self.crust_nodes.push(new_node);
+      return true;
+    }
+    return false;
+  }
+
+  pub fn drop_node(&mut self, lost_node : CrustNode) -> bool {
+    for node in self.crust_nodes.iter_mut() {
+      if node.endpoint == lost_node.endpoint {
+        node.set_disconnected();
+        return true;
+      }
+    }
+    return false;
+  }
+
+  pub fn get_connected_nodes(&self) -> Vec<CrustNode> {
+    self.crust_nodes.iter()
+                    .filter_map(|node| if node.connected == true {
+                      Some(node.clone())
+                      } else {None})
+                    .collect::<Vec<_>>()
+  }
+
+  pub fn get_all_nodes(&self) -> Vec<CrustNode> {
+    self.crust_nodes.clone()
+  }
+}
+
+fn main() {
+
+  let args : Args = Docopt::new(USAGE)
+                     .and_then(|d| d.decode())
+                     .unwrap_or_else(|e| e.exit());
+
+  if !args.flag_help { println!("{:?}", args); };  // TODO: remove; here for debug
+  if args.flag_help {
+    println!("{:?}", args);     // print help message
+    return;
+  };
+
+  let (cm_tx, cm_rx) = channel();
+  let cm = ConnectionManager::new(cm_tx);
+  let cm_eps = match cm.start_listening(vec![Port::Tcp(5483u16)]) {
+    Ok(eps) => eps,
+    Err(e) => panic!("Connection manager failed to start on arbitrary TCP port: {}", e)
+  };
+  assert!(cm_eps.len() > 0);
+  for ep in &cm_eps {
+    match *ep {
+      Endpoint::Tcp(socket) => println!("Connection manager now listening on TCP socket {}", socket)
+    };
+  };
+  let mut my_flat_world : FlatWorld = FlatWorld::new(cm_eps);
+
+  let mut default_bootstrap = !args.flag_bootstrap;
+  if args.flag_bootstrap {
+    match args.arg_peer {
+      Some(peer) => {
+        // String.as_str() is unstable; waiting RFC revision
+        // http://doc.rust-lang.org/nightly/std/string/struct.String.html#method.as_str
+        let bootstrap_address = match SocketAddr::from_str(peer.as_str()) {
+          Ok(addr) => addr,
+          Err(_) => panic!("Failed to parse bootstrap peer as valid IPv4 or IPv6 address: {}", peer)
+        };
+        match cm.bootstrap(Some(vec![Endpoint::Tcp(bootstrap_address)])) {
+          Ok(endpoint) =>  my_flat_world.add_node(CrustNode::new(endpoint, true)),
+          Err(e) => { println!("Failed to bootstrap from provided peer: {}, with error : {}", peer, e);
+                      panic!("Not resulting to default discovery of bootstrap nodes. Exiting"); }
+                      // default_bootstrap = true; }
+        };
+      },
+      None => { println!("No peer address provided, resort to default");
+                default_bootstrap = true; }
+    }
+  }
+
+  // resort to default bootstrapping methods
+  if default_bootstrap && !args.flag_origin {
```

Comment:
Yes, that's the purpose to allow skip all bootstrapping attempts.
Because for the first node, there is no point trying to bootstrap.